Sources/Sentry/SentryWatchdogTerminationTrackingIntegration.m

```diff
@@ -80,13 +84,26 @@ - (BOOL)installWithOptions:(SentryOptions *)options
 
     self.appStateManager = appStateManager;
 
+    // Setup the scope observer with its data processors
+    SentryWatchdogTerminationBreadcrumbProcessor *breadcrumbProcessor =
+        [SentryDependencyContainer.sharedInstance
+            getWatchdogTerminationBreadcrumbProcessorWithMaxBreadcrumbs:options.maxBreadcrumbs];
+    SentryWatchdogTerminationContextProcessor *contextProcessor =
+        [SentryDependencyContainer.sharedInstance watchdogTerminationContextProcessor];
+
     SentryWatchdogTerminationScopeObserver *scopeObserver =
         [[SentryWatchdogTerminationScopeObserver alloc]
-            initWithMaxBreadcrumbs:options.maxBreadcrumbs
-                       fileManager:[[[SentrySDK currentHub] getClient] fileManager]];
+            initWithBreadcrumbProcessor:breadcrumbProcessor
+                       contextProcessor:contextProcessor];
+
+    [SentrySDK.currentHub configureScope:^(SentryScope *_Nonnull outerScope) {
+        // Add the observer to the scope so that it can be notified when the scope changes.
+        [outerScope addObserver:scopeObserver];
 
-    [SentrySDK.currentHub configureScope:^(
-        SentryScope *_Nonnull outerScope) { [outerScope addObserver:scopeObserver]; }];
+        // Sync the current context to the observer to capture context modifications that happened
+        // before installation.
+        [scopeObserver setContext:outerScope.contextDictionary];
```

Comment:
`m`: If I'm not mistaken, no tests validate this functionality. It's a bit suspicious that we have no changes in `SentryWatchdogTerminationIntegrationTests`.